File: 2441-largest-positive-integer-that-exists-with-its-negative/2441-largest-positive-integer-that-exists-with-its-negative.java
class Solution {
    public int findMaxK(int[] nums) {
        Arrays.sort(nums);
        int lo = 0, hi = nums.length - 1;

        while (lo < hi) {
            if (nums[lo] == -1 * nums[hi]) {
                return nums[hi];
            }
            if (nums[lo] < -1 * nums[hi]) {
                lo++;
            } else {
                hi--;
            }
        }

        return -1;
    }
}

File: 2441-largest-positive-integer-that-exists-with-its-negative/2441-largest-positive-integer-that-exists-with-its-negative.py
class Solution:
    def findMaxK(self, nums: List[int]) -> int:
        nums.sort()
        lo, hi = 0, len(nums) - 1

        while lo < hi:
            if nums[lo] == -1 * nums[hi]:
                return nums[hi]
            if nums[lo] < -1 * nums[hi]:
                lo += 1
            else:
                hi -= 1

        return -1